src/webapp/src/characters/Characters.tsx

```diff
@@ -0,0 +1,39 @@
+import React from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { RootState } from '../reducers';
+import { getCharacters } from './charactersSlice';
+
+import './Characters.css';
+
+export const Characters = () => {
+  const dispatch = useDispatch();
+  const { results, error } = useSelector(
+    (state: RootState) => state.characters
+  );
+
+  const charactersGrid = results.map(({ name, species, image, id }) => (
+    <div key={id} className='card'>
+      <div className='picture-wrap'>
+        <img src={image} title={name} alt='character' />
+      </div>
+      <div className='card-footer'>
+        <h3 className='character-name'>{name}</h3>
+        <div>{species}</div>
+      </div>
+    </div>
+  ));
```

Comment:
Hi @paschalidi, I was specially looking forward to listen to your opinion in this topic. I have found a lot of different opinions and I am still trying to decided what _I_ personally find more readable, like you said, it is a very opinionated matter. 

Having said that, I made this abstraction thinking of you (that didn't work :)) because I am also inclined to think it is better to put the code directly in the `return` statement. So I changed it and I am fine with following this pattern as long as it makes sense. If not, I will make it a new component like you suggested (I just thought that in this case making a new component was unnecessary)

Reply:
>(I just thought that in this case making a new component was unnecessary) 

It might be yes. Sometimes we make components not because they are reusable but to hide complexity. Like if you have a very complicated component that in the end displays some sort of list, then you might wanna create a component with a descriptive name. Then you will hide the complexity from the reader. Now if the reader is interested in that part they will open the component and start reading it. If not they will move on. 

Not this isnt very complex here, imo.